File: src/main/java/bad/robot/radiate/teamcity/FullProject.java
package bad.robot.radiate.teamcity;

import com.google.gson.annotations.SerializedName;

import java.util.Iterator;

@Deprecated
class FullProject extends Project {

    @SerializedName("buildTypes")
    private final BuildTypes buildTypes;

    public FullProject(String id, String name, String href, BuildTypes buildTypes) {
        super(id, name, href);
        this.buildTypes = buildTypes;
    }

    @Override
    public Iterator<BuildType> iterator() {
        return buildTypes.iterator();
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/FullProject.scala
package bad.robot.radiate.teamcity

import argonaut.Argonaut._
import argonaut.DecodeJson
import bad.robot.radiate.HypermediaS

object ProjectScala {
  implicit def fullProjectDecoder: DecodeJson[ProjectScala] =
    DecodeJson(cursor => {
      for {
        id <- (cursor --\ "id").as[String]
        name <- (cursor --\ "name").as[String]
        href <- (cursor --\ "href").as[String]
        buildTypes <- (cursor --\ "buildTypes").as[BuildTypesScala]
      } yield {
        new ProjectScala(id, name, href, buildTypes)
      }
    })
}

case class ProjectScala(id: String, name: String, href: String, buildTypes: BuildTypesScala) extends TeamCityObjectS with HypermediaS {
  override def toString = s"$name ($id)"
}